### PR TITLE
Dg/344 invalid permission

### PR DESCRIPTION
**Pull request by Dimon70007**
# Summary
Added info about ability of using this library with `use_frameworks!` tag

Sometimes we could not use modular_headers because of dependencies with precompiled dynamic frameworks. as workaround - we can manually specify dependency type for all dependencies, but dynamic frameworks
### What's required for testing (prerequisites)?
This approach will work only on cocoapods 1.7+ version
## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [ ] I have tested this on a device and a simulator
+ [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
+ [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by zoontek** [comment]
@Dimon70007 Hi, and thanks for the PR!

An alternative workaround could be https://github.com/react-native-community/react-native-permissions/issues/348#issuecomment-541423154. What do you think of it?

**Comment by Dimon70007** [comment]
Hi @zoontek. I have read https://github.com/react-native-community/react-native-permissions/issues/348#issuecomment-541423154 and think that it is too difficult for users to manually link every static library. Even more, i have tried to use this approach (manual linking), but have no luck

File: example/ios/Podfile [hunk 1]
```diff
@@ -54,3 +74,13 @@ target 'RNPermissionsExample' do
 
   use_native_modules!
 end
+
+# post_install do |installer|
```
Comment (zoontek) [inline example/ios/Podfile]: Id this needed in any way or can be removed?

File: tsconfig.json [hunk 2]
```diff
@@ -2,10 +2,10 @@
   "compilerOptions": {
     "esModuleInterop": true,
     "jsx": "react-native",
-    "lib": ["es2015"],
+    "lib": ["es2017"],
```
Comment (zoontek) [inline tsconfig.json]: Why the changes?

File: RNPermissions.podspec [hunk 3]
```diff
@@ -1,5 +1,5 @@
 require 'json'
-package = JSON.parse(File.read('./package.json'))
+package = JSON.parse(File.read(File.join(__dir__, 'package.json')))
```
Comment (zoontek) [inline RNPermissions.podspec]: Why changing this?

File: example/ios/Podfile [hunk 4]
```diff
@@ -1,4 +1,24 @@
 platform :ios, '9.0'
+use_frameworks!
+# use_modular_headers!
+
+dynamic_frameworks = []
+
+# make all the other frameworks into static frameworks by overriding the static_framework? function to return true
```
Comment (zoontek) [inline example/ios/Podfile]: Not sure it's a good idea to make the example uses `use_frameworks!` be default. It could scare beginners who won't have this issue anyway.

File: example/ios/Podfile (hunk 1 above)
Comment (Dimon70007) [reply example/ios/Podfile]: oh, sorry
i have forgot to remove dead code
i will do new PR soon

File: tsconfig.json (hunk 2 above)
Comment (Dimon70007) [reply tsconfig.json]: this changes needed for compatibility with es2017 - `Object.values` syntax, used in src/index.ts

Comment (zoontek) [reply tsconfig.json]: That's not useful since Metro + react-native-babel-preset can take care of it anyway :)

File: example/ios/Podfile (hunk 4 above)
Comment (Dimon70007) [reply example/ios/Podfile]: ok, but if the beginner will copy this to his podfile - it will break nothing.

File: RNPermissions.podspec (hunk 3 above)
Comment (Dimon70007) [reply RNPermissions.podspec]: ok. i'll drop this

File: example/ios/Podfile (hunk 4 above)
Comment (zoontek) [reply example/ios/Podfile]: Of course not. But the users who don't need it don't have to think this is mandatory. Commenting the whole block + adding a similar comment as in the `README` could be a great solution.

Comment (Dimon70007) [reply example/ios/Podfile]: ok. done

**Comment by zoontek** [comment]
Thanks!
